File: problems/452_minimum_number_of_arrows_to_burst_balloons.py
Point = list[int]


class Solution:
    def findMinArrowShots(self, points: list[Point]) -> int:
        points.sort(key=lambda p: p[0])

        current_range: int | None = None
        count = 0

        for point in points:
            if current_range is None or point[0] > current_range:
                count += 1
                current_range = point[1]
            else:
                current_range = min(current_range, point[1])

        return count

File: problems/452-minimum-number-of-arrows-to-burst-balloons.ts
type Point = number[]

function findMinArrowShots(points: Point[]): number {
  // Sort by start positions.
  points.sort((a, b) => a[0] - b[0])

  let currentRange: number | undefined
  let count = 0

  for (const point of points) {
    if (currentRange === undefined || point[0] > currentRange) {
      count++
      currentRange = point[1]
    } else {
      currentRange = Math.min(currentRange, point[1])
    }
  }

  return count
}
